src/drivers/display/drm/lv_linux_drm.c

```diff
@@ -783,11 +884,117 @@ static int drm_allocate_dumb(drm_dev_t * drm_dev, drm_buffer_t * buf)
     return 0;
 }
 
+#if LV_LINUX_DRM_GBM_BUFFERS
+
+static int create_gbm_buffer(drm_dev_t * drm_dev, drm_buffer_t * buf)
+{
+
+    struct gbm_bo * gbm_bo;
+    int prime_fd;
+    uint32_t handles[4] = {0}, pitches[4] = {0}, offsets[4] = {0};
+    uint32_t w, h, format;
+    uint32_t n_planes;
+    char * drm_format_name;
+    int res;
+
+    /* gbm_bo_format does not define anything other than ARGB8888 or XRGB8888 */
+    if(LV_COLOR_DEPTH != 32) {
+        LV_LOG_ERROR("Unsupported color format");
+        return -1;
+    }
+
+    /* Create a linear GBM buffer object - best practice when modifiers are not used */
+    if(!(gbm_bo = gbm_bo_create(gbm_device,
+                                drm_dev->width, drm_dev->height, GBM_BO_FORMAT_XRGB8888,
+                                GBM_BO_USE_SCANOUT | GBM_BO_USE_LINEAR))) {
+
+        LV_LOG_ERROR("Unable to create gbm buffer object");
+        return -1;
+    }
+
+    /* Currently only, one plane per dma-buf/prime fd is supported - but some GPUs feature
+     * several planes (multiple fds or sometimes a single fd for multiple planes).
+     * current implementation is kept simple for now */
+
+    n_planes = gbm_bo_get_plane_count(gbm_bo);
+
+    if(n_planes != 1) {
+        LV_LOG_ERROR("The current implementation only supports a single plane per fd");
+        return -1;
+    }
+
+    w = gbm_bo_get_width(gbm_bo);
+    h = gbm_bo_get_height(gbm_bo);
+    format = gbm_bo_get_format(gbm_bo);
+    pitches[0] = buf->pitch = gbm_bo_get_stride_for_plane(gbm_bo, 0);
+    offsets[0] = buf->offset = gbm_bo_get_offset(gbm_bo, 0);
+    buf->size = h * buf->pitch;
+    drm_format_name = drmGetFormatName(format);
+
+    LV_LOG_INFO("Created GBM BO of size: %lu pitch: %u offset: %u format: %s",
+                buf->size, buf->pitch, buf->offset, drm_format_name);
+
+    prime_fd = gbm_bo_get_fd_for_plane(gbm_bo, 0);
+
+    if(prime_fd < 0) {
+        LV_LOG_ERROR("Failed to get prime fd for plane 0");
+        return -1;
+
+    }
+
+    buf->map = mmap(NULL, buf->size, PROT_READ | PROT_WRITE, MAP_SHARED, prime_fd, 0);
```

Comment:
Should it be `munmap`ped anywhere or does it live for the duration of the program?